src/wstool/config_yaml.py

```diff
@@ -378,29 +382,39 @@ def get_path_spec_from_yaml(yaml_dict):
 
 
 def generate_config_yaml(config, filename, header, pretty=False,
-                         sort_with_localname=False):
+                         sort_with_localname=False, curr_revision=False,
+                         vcs_only=False):
     """
     Writes file filename with header first and then the config as yaml
     """
     if not os.path.exists(config.get_base_path()):
         os.makedirs(config.get_base_path())
-    config_filepath = os.path.realpath(os.path.join(config.get_base_path(), filename))
-    with open(config_filepath, 'w+b') as f:
-        if header is not None:
-            f.write(header.encode('UTF-8'))
-        if sort_with_localname:
-            items = [x.get_legacy_yaml() for x in
-                        sorted(config.get_source(),
-                               key=lambda x:x.get_local_name())]
-        else:
-            items = [x.get_legacy_yaml() for x in config.get_source()]
 
-        if not items:
-            return
-
-        if pretty:
-            content = yaml.safe_dump(items, allow_unicode=True,
-                                     default_flow_style=False)
-        else:
-            content = yaml.safe_dump(items)
-        f.write(content.encode('UTF-8'))
+    content = ""
+    if header:
+        content += header
+    if sort_with_localname:
+        items = [x.get_legacy_yaml(curr_revision) for x in
+                 sorted(config.get_source(curr_revision, vcs_only),
+                        key=lambda x:x.get_local_name())]
+    else:
+        items = [x.get_legacy_yaml(curr_revision)
+                 for x in config.get_source(curr_revision, vcs_only)]
+
+    if not items:
+        return
```

Comment:
I think this causes the behavior to change. Before, when you reached this case, the file was already opened and the header was already written. Now the file will not have been opened and the header will not have been written. I don't know which behavior is correct off-hand, but this is definitely different from before. It might be easier to avoid restructuring this logic to make the change as minimal as possible.